lib/exceptions.rb

```diff
@@ -170,6 +170,7 @@ class UpgradeFailed < MachineryError; end
     class BinaryDiffError < MachineryError; end
 
     class SshConnectionFailed < MachineryError; end
+    class SshConnectionDisrupted < SystemExit; end
```

Comment:
The probability of an error of this sort occurring while inspecting is already very low, i had to manually kill the tcp connection and shut down the sshd deamon to even replicate it. If it does get back up again you can try again, otherwise you would have to fix the connection first anyway. The error mentioned in the bug report isn't a timeout but rather a configuration problem because the machine shuts down the connection on purpose.